src/kernel/include/thd.h

```diff
@@ -520,26 +585,60 @@ thd_preemption_state_update(struct thread *curr, struct thread *next, struct pt_
 	memcpy(&curr->regs, regs, sizeof(struct pt_regs));
 }
 
+static int
```

Comment:
We didn't talk about all of this. The SCB is also used to send interrupt events up to user-level now.